xsamtools/cram.py

```diff
@@ -51,6 +51,54 @@ def read_fixed_length_cram_file_definition(fh: io.BytesIO):
+    | Data Type      Name                        Value                                                        |
+    -----------------------------------------------------------------------------------------------------------
+    | INT32          length                      The sum of the lengths of all blocks in this container       |
+    |                                              (headers and data); equal to the total byte length of the  |
+    |                                              container minus the byte length of this header structure.  |
+    | ITF-8           reference sequence id      Reference sequence identifier or:                            |
+    |                                               -1 for unmapped reads                                     |
+    |                                               -2 for multiple reference sequences                       |
+    |                                            All slices in this container must have a reference sequence  |
+    |                                              id matching this value.                                    |
+    | ITF-8           reference start position   The alignment start position or 0 if the container is        |
+    |                                              multiple-reference or contains unmapped unplaced reads     |
+    | ITF-8           alignment span             The length of the alignment or 0 if the container is         |
+    |                                              multiple-reference or contains unmapped unplaced reads.    |
+    | ITF-8           number of records          Number of records in the container.                          |
+    | LTF-8           record counter             1-based sequential index of records in the file/stream.      |
+    | LTF-8           bases                      Number of read bases.                                        |
+    | ITF-8           number of blocks           The total number of blocks in this container.                |
+    | Array<ITF-8>    landmarks                  The locations of slices in this container as byte offsets    |
+    |                                             from the end of this container header, used for random      |
+    |                                             access indexing. The landmark count must equal the slice    |
+    |                                             count.  Since the block before the first slice is the       |
+    |                                             compression header, landmarks[0] is equal to the byte       |
+    |                                             length of the compression header.                           |
+    | INT             crc32                      CRC32 hash of the all the preceding bytes in the container.  |
+    -----------------------------------------------------------------------------------------------------------
```

Comment:
This docstring is beautiful. But if it’s similar to the CRAM spec I’d be tempted to use a link to avoid maintaining it.

Reply:
Hmmm... noted and a good point.  I'd like to leave it in though.  >.>  It hasn't been updated in six years so I don't mind updating this upon the next release (if and when it occurs).